### PR TITLE
Verify physical file is linked and exactly 1 created in lakefs create test

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

File: clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java [hunk 1]
```diff
@@ -141,6 +151,27 @@ public void setUp() throws Exception {
...
+    /**
+     * @return all pathnames under s3Prefix that start with prefix.  (Obvious not scalable!)
+     */
+    protected List<String> getS3FilesByPrefix(String prefix) throws IOException {
+        final int maxKeys = 1500;
+
+        ListObjectsRequest req = new ListObjectsRequest().
+            withBucketName(s3Bucket).
+            withPrefix(prefix)
+            .withMaxKeys(maxKeys);
```
Comment (nopcoder) [inline clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java:163]: java
```suggestion
        ListObjectsRequest req = new ListObjectsRequest()
                .withBucketName(s3Bucket)
                .withPrefix(prefix)
                .withMaxKeys(maxKeys);
```

File: clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java [hunk 2]
```diff
@@ -141,6 +151,27 @@ public void setUp() throws Exception {
...
+        ObjectListing listing = s3Client.listObjects(req);
+        if (listing.isTruncated()) {
+            Assert.fail(String.format("[internal] no support for test that creates >%d S3 objects", maxKeys));
+        }
+
+        return listing.getObjectSummaries().
+            stream().
+            map(summary -> summary.getKey()).
+            collect(Collectors.toList());
```
Comment (nopcoder) [inline clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java:172]: Consider
```java
return Lists.transform(listing.getObjectSummaries(), S3ObjectSummary::getKey);
```

**Review by nopcoder — APPROVED** [review]
minor suggestions

File: clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java (hunk 1 above)
Comment (arielshaqed) [reply clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java]: Done.  The [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break) even says explicitly to do it like this even though "_this is not the same practice used in Google style for other languages, such as C++ and JavaScript_"...

Emacs-Lisp command for further reference "if" it happened elsewhere:
```emacs-lisp
(query-replace-regexp "\\.
\\( *\\)" "
\\1." nil nil nil nil nil)
```

File: clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java (hunk 2 above)
Comment (arielshaqed) [reply clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java]: Yikes, thanks!  Forgot that (lazy) "map" is called "transform" in *Google*'s Java library!

Also thanks for reminding me of the one place Java forgot its syntax and decided to use C++ `::`.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Comment by arielshaqed** [comment]
Code change unrelated to Nessie test failure, so pulling by administrative force.

**Pull request merged**